sdk/storage/azfile/share/client_test.go

```diff
@@ -0,0 +1,1343 @@
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	shareName := testcommon.GenerateShareName(testName)
+	shareClient := testcommon.CreateNewShare(context.Background(), _require, shareName, svcClient)
+	defer testcommon.DeleteShare(context.Background(), _require, shareClient)
+
+	newQuota := int32(300)
+
+	// In order to test and get LastModified property.
+	_, err = shareClient.SetProperties(context.Background(), &share.SetPropertiesOptions{Quota: to.Ptr(newQuota)})
+	_require.NoError(err)
+
+	gResp, err := shareClient.GetStatistics(context.Background(), nil)
+	_require.NoError(err)
+	_require.Equal(gResp.Date.IsZero(), false)
+	// _require.NotEqual(*gResp.ETag, "") // TODO: The ETag would be ""
+	// _require.Equal(gResp.LastModified.IsZero(), false) // TODO: Even share is once updated, no LastModified would be returned.
+	_require.NotNil(gResp.RequestID)
+	_require.NotNil(gResp.Version)
+	_require.Equal(*gResp.ShareUsageBytes, int32(0))
+}
+
+func (s *ShareUnrecordedTestsSuite) TestShareGetStatsNegative() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	shareName := testcommon.GenerateShareName(testName)
+	shareClient := testcommon.GetShareClient(shareName, svcClient)
+
+	_, err = shareClient.GetStatistics(context.Background(), nil)
+	_require.Error(err)
+	testcommon.ValidateFileErrorCode(_require, err, fileerror.ShareNotFound)
+}
+
+// TODO: uncomment this test after directory and file clients are added
+/*func (s *ShareUnrecordedTestsSuite) TestSetAndGetStatistics() {
+	_require := require.New(s.T())
+	testName := s.T().Name()
+	svcClient, err := testcommon.GetServiceClient(s.T(), testcommon.TestAccountDefault, nil)
+	_require.NoError(err)
+
+	shareName := testcommon.GenerateShareName(testName)
+	shareClient := testcommon.GetShareClient(shareName, svcClient)
+
+	_, err = shareClient.Create(context.Background(), &share.CreateOptions{Quota: to.Ptr(int32(1024))})
+	_require.NoError(err)
+	defer testcommon.DeleteShare(context.Background(), _require, shareClient)
+
+	dirClient := shareClient.NewDirectoryClient("testdir")
+	_, err = dirClient.Create(context.Background(), nil)
+	_require.NoError(err)
+
+	fileClient := dirClient.NewFileClient("testfile")
+	_, err = fileClient.Create(context.Background(), int64(1024 * 1024 * 1024 * 1024), nil)
+	_require.NoError(err)
+
+	getStats, err := shareClient.GetStatistics(context.Background(), nil)
+	_require.NoError(err)
+	_require.Equal(*getStats.ShareUsageBytes, int64(1024*1024*1024*1024))
+}*/
+
+func deleteShare(ctx context.Context, _require *require.Assertions, shareClient *share.Client, o *share.DeleteOptions) {
+	_, err := shareClient.Delete(ctx, o)
+	_require.NoError(err)
+}
+
+func (s *ShareUnrecordedTestsSuite) TestShareCreateSnapshotNonDefault() {
```

Comment:
Should some of these tests be recorded? Or is the idea to record all at once in the end?

Reply:
Yes, will record the tests in some future PR.